core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala

```diff
@@ -71,14 +71,17 @@ private[spark] class StreamingPythonRunner(
     val prevConf = conf.get(PYTHON_USE_DAEMON)
     conf.set(PYTHON_USE_DAEMON, false)
     try {
-      val (worker, _) = env.createPythonWorker(
-        pythonExec, workerModule, envVars.asScala.toMap)
+      val workerFactory =
+        new PythonWorkerFactory(pythonExec, workerModule, envVars.asScala.toMap)
+      val (worker: PythonWorker, _) = workerFactory.createSimpleWorker(blockingMode = true)
```

Comment:
What is this change about? 
It broke `stop()` method below. 
cc: @WweiL, @HyukjinKwon

Reply:
Yes it breaks the stop() method below. It should be updated to like this: 
https://github.com/apache/spark/blob/branch-3.5/core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala#L109-L113

Reply:
@utkarsh39 we will create a followup ticket to fix this

Reply:
Thanks @WweiL